doc/read-the-docs-site/adr/0006-indexer-resuming-strategy.rst

```diff
@@ -45,16 +45,88 @@ Decision
+    -- have the same lowest immutable point, select the one which has the lowest
+    -- overall point (immutable or not).
+    selectIntersectionPoints indexers = do
+      resumingPoints <- mapM queryResumingPoints indexers
+      head $ sortOn (\rps -> (maximum $ filter isImmutablePoint rps, maximum rps)) resumingPoints
+
+  Next, we need a function which processes each ``ChainSyncEvent`` returned by the chain-sync client for each indexer.
+  The pseudo-code is as follows:
+
+  .. code-block:: haskell
+
+    applyChainSyncEvent event indexer = do
+      case event of
+        lastSyncPoints <- getLastSyncPoints indexer
+        Resume point -> do
+          maybePoint <- getPointFromSlotNo (slotNo point) indexer
+          case maybePoint of
+            Nothing ->
+              pure () -- Do nothing
+            Just _ ->
+              rewind point indexer
+        RollBackward point -> do
+          rewind point indexer
+        RollForward block point -> do
+          lastSyncPoints <- getLastSyncPoints indexer
+          let lastSyncSlotNos = fmap getSlotNo lastSyncPoints
+          let lowestLastSyncSlotNo = minimum lastSyncSlotNos
+          let highestLastSyncSlotNo = maximum lastSyncSlotNos
+
+          -- Block already processed. Do nothing
+          if point `elem` lastSyncPoints
+          then do
+            pure ()
+
+          -- SlotNo of new block point already processed, but with different block header hash
+          else if getSlotNo point `elem` lastSyncSlotNos
+          then do
+            -- TODO Not exactly correct, we need to rewind to `point - 1`, but we don't necessarly have
+            -- the block header hash of that point
+            rewind point indexer
+            insert block indexer
+
+          -- Block already processed. Do nothing
+          else if getSlotNo point < lowestLastSyncSlotNo
+          then do
+            pure ()
+
+          -- New block not processed yet.
+          else if getSlotNo point > highestLastSyncSlotNo
+          then do
+            insert block indexer
+
+          -- Should never go here
+          else do
+            pure ()
+
+  It is important to note that the implementation of ``applyChainSyncEvent`` is highly depend on the implementation of ``selectIntersectionPoints``.
 
 Argument
 --------
 
-In order to justify the decision, we will present various use case scenarios and show how the decisions satisfies them.
+The reason ``resumeFromStorage`` returns a list of resuming points where there is *at least* one immutable point is to ensure we can resume from an existing chain point when restarting Marconi.
+Then, the reason why we include rollbackable points in the list of resuming points is simply to resume from a point close to the tip, thus prevent re-indexing of already indexed information.
+
+The main drawback of the solution is that each indexer must keep track of the last sync points.
```

Comment:
Add a note that all the solutions written so far relied on this assumption, although it wasn't always clearly stated.